Enable High-Performance GPU Selection for Windows in JVM

mo-beck opened the pull request:
**Work Item**: `1949141`  
**Metadata file**: `ms-patches/enable-high-performance-gpu-selection-for-windows-in-jvm.yml`  

### Summary
This patch exports two vendor-specific variables during JVM startup on Windows to hint GPU drivers to favor discrete GPUs (when available):
- `NvOptimusEnablement` for NVIDIA
- `AmdPowerXpressRequestHighPerformance` for AMD

### Motivation 
Improves performance in scenarios where discrete GPU usage is beneficial, including:
- Games (e.g., Minecraft)
- Machine learning tools
- IDEs or GUI-intensive applications

karianna commented:
Is there anyway to add a test for this?

mo-beck commented:
@karianna: I've also reached out to the Minecraft team to confirm whether they'd prefer this to be gated behind a flag (e.g., `-XX:+EnableHighPerformanceGPUHint`). If so, I can certainly add a guard and write up a test. Will follow up once I hear back from them.

mo-beck commented:
> @karianna: I've also reached out to the Minecraft team to confirm whether they'd prefer this to be gated behind a flag (e.g., `-XX:+EnableHighPerformanceGPUHint`). If so, I can certainly add a guard and write up a test. Will follow up once I hear back from them.

Chloe responded that “it sounds perfect” as-is and that they don’t anticipate a need to ever disable the exports. She added there's no objection to having a flag, but no strong need for one either.

So I’ll leave the implementation unguarded for now — minimal and default-enabled. If upstream reviewers later request a flag for configurability, I’m happy to add one. /cc @macarte

macarte reviewed (approved):
LGTM

karianna commented:
Not sure how this merges?

brianjstafford commented:
> Not sure how this merges?

This actually isn't supposed to merge into main. The ms-patches branch that has been created will be merged along with other ms-patches branches at release time. I think what is supposed to happen (and we have no way to enforce it), is that an ms-patches PR should be *from* a person's dev branch for the feature *to* the ms-patches/<feature_branch> branch. The latter branch would initially just be a copy from main. That way when the PR is approved and merged the ms-patches branch is set up properly with the intended additional patch. In this case, the ms-patches branch was already created before the PR was. I think, since we've gotten the necessary reviews, that the PR can simply be closed without integrating to main. If it was rejected, we would need to delete the branch to avoid it getting included in a release.

macarte commented:
@mo-beck - you'll also need to update the ms-patches index: https://github.com/microsoft/openjdk-jdk21u/tree/ms-patches/microsoft-patch-index/ms-patches

that's in the ms-patches/microsoft-patch-index branch in the ms-partches folder

mo-beck commented:
> @mo-beck - you'll also need to update the ms-patches index: https://github.com/microsoft/openjdk-jdk21u/tree/ms-patches/microsoft-patch-index/ms-patches
> 
> that's in the ms-patches/microsoft-patch-index branch in the ms-partches folder

Done: https://github.com/microsoft/openjdk-jdk21u/pull/33

mo-beck commented:
> > Not sure how this merges?
> 
> This actually isn't supposed to merge into main. The ms-patches branch that has been created will be merged along with other ms-patches branches at release time. I think what is supposed to happen (and we have no way to enforce it), is that an ms-patches PR should be _from_ a person's dev branch for the feature _to_ the ms-patches/<feature_branch> branch. The latter branch would initially just be a copy from main. That way when the PR is approved and merged the ms-patches branch is set up properly with the intended additional patch. In this case, the ms-patches branch was already created before the PR was. I think, since we've gotten the necessary reviews, that the PR can simply be closed without integrating to main. If it was rejected, we would need to delete the branch to avoid it getting included in a release.

Thanks @brianjstafford . I had been following the internal [ms-patches process doc](https://microsoft.sharepoint.com/:w:/t/Juniper/ESh-ByREXtlCgh0sl03kqvkB2lOrdNPra2DIoe5xt2oEYA?e=BPbfQj ) which doesn't yet make the "PR from dev branch to ms-patches/<feature>" model totally clear. That led to PR #32 being created with the assumption that it needed to land in main, when in fact the `ms-patches/enable-high-performance-gpu-selection-for-windows-in-jvm` branch was already the correct home for the patch. 

I will go ahead and close this PR as requested. I have added the patch metadata and updated central index in pr #33, which targets the `ms-patches/microsoft-patch-index` branch as expected.

I think everything is now correctly wired up, so that the patch will be included in future release builds. I'll also work on improving the doc so that this nuance is clearer going forward. Thanks again for your help getting this sorted!
